### PR TITLE
Vs2 settings

### PR DESCRIPTION
**Pull request by BIGduzy**

### CODE REVIEW

File: src/hc-06.hpp [hunk 1]
```diff
@@ -1,24 +1,163 @@
...
+    hwlib::string<pinSize> pincode;       ///< Used for storing a local version of the 4 digit pincode saved as a byte
   private:
-    uint8_t discoveredDevices[32]; ///< Used for storing the connection id of a discovered device.
-    unsigned int currentBaudrate;
+    /*
+     * @brief convert uint32_t to hwlib::string
+     *
+     * Removes 0's before the first non zero value
+     *
+     * @param The number that needs to be converted
```
Comment (DanielvanVliet) [inline src/hc-06.hpp]: incorrect doxygen

File: src/hc-06.hpp [hunk 2]
```diff
@@ -1,24 +1,163 @@
...
+     *
+     * Uses the commandType to create a command with the given data, then sends it
+     * The uses the reveive method to check for response of the decive and checks if the command was successfully set
+     *
+     * @tparam     size          Size of the data string
+     * @param      commandType   The command type of the command that needs to be send
+     * @param      data          The data that needs to be send
+     * @return If the command was successfully received at the device
+     */
+    template <size_t size, size_t responseSize>
```
Comment (ghost) [inline src/hc-06.hpp:128]: Missing a @tparam command for responseSize.

**Review by DanielvanVliet — CHANGES_REQUESTED** [review]
Small doxygen fix, looks good otherwise

File: src/hc-06.hpp [hunk 3]
```diff
@@ -1,24 +1,163 @@
...
-    unsigned int currentBaudrate;
+    /*
+     * @brief convert uint32_t to hwlib::string
+     *
+     * Removes 0's before the first non zero value
+     *
+     * @param The number that needs to be converted
+     * @return The number in string format
+     */
+    hwlib::string<0> uint32_tToString(uint32_t number) {
```
Comment (ghost) [inline src/hc-06.hpp]: Missing @param command for number.

File: src/hc-06.hpp [hunk 4]
```diff
@@ -1,24 +1,163 @@
...
+        }
+        return nonZeroString;
+    }
+
+    /*
+     * @brief compares to hwlib::strings
+     *
+     * @return True if equal, false if not
+     */
+    template <size_t size>
```
Comment (ghost) [inline src/hc-06.hpp]: Missing @tparam for size.

File: src/hc-06.hpp [hunk 5]
```diff
@@ -41,16 +180,33 @@ class HC06 : public Bluetooth {
...
      */
-    hwlib::string<50> getName();
+    hwlib::string<maxNameSize> getName();
 
     /**
-     * @brief Used to read the status of the HC-06 chip.
+     * @brief Used to set the name of the chip. This is not an ID
      *
-     * This function will check if the chip is detected and functional by performing a simple communication test.
+     * @param[in]     name    A reference to a string holding the name.
```
Comment (ghost) [inline src/hc-06.hpp]: Should be newName instead of name.

**Review by ghost — CHANGES_REQUESTED** [review]
Some doxygen things have to be fixed, but besides those, everything looks good.

File: src/hc-06.hpp (hunk 1 above)
Comment (BIGduzy) [reply src/hc-06.hpp]: I don't see what I do wrong here

**Review by ghost — APPROVED** [review]

**Review by DanielvanVliet — APPROVED** [review]

**Pull request merged**